### ISSUE TITLE
Nextcloud 3.1.0 (from RC1) insta-crashed on Pop_OS 20.10

### ISSUE THREAD

**Issue by cybik** [nextcloud/desktop#2737]
### Expected behaviour
No Crash

### Actual behaviour
Segfault on 3.1.0. Stacktrace in Logs

### Steps to reproduce
1. Have a user account configured with a previous Nextcloud client (way, WAY old maybe)
2. Install 3.1.0 from either the PPA or using the AppImage releases
3. Explode

### Client configuration
Client version: Fails on 3.1.0, but 3.0.3 is just fine
Operating system: Pop_OS 20.10 (Linux distribution)
OS language: Multiple (FR, EN)
Qt version used by client package (Linux only, see also Settings dialog): 5.14
Client package (From Nextcloud or distro) (Linux only): both NextCloud Appimage and PPA
Installation path of client: either `/usr/bin` or `/opt/nextcloud`, no difference

### Server configuration
Nextcloud version: 20.0.3
Storage backend (external storage): flat filesystem

### Logs

```
Thread 1 "nextcloud" received signal SIGSEGV, Segmentation fault.
0x00005555556f08fe in OCC::User::avatar() const ()
(gdb) st
start     starti    step      stepi     stepping  stop      strace    
(gdb) backtrace 
#0  0x00005555556f08fe in OCC::User::avatar() const ()
#1  0x00005555556f09ff in OCC::UserModel::avatarById(int const&) ()
#2  0x00005555556f1798 in OCC::ImageProvider::requestImage(QString const&, QSize*, QSize const&) ()
#3  0x00007fffecff2bd0 in QQuickPixmap::load(QQmlEngine*, QUrl const&, QSize const&, QFlags<QQuickPixmap::Option>, QQuickImageProviderOptions const&, int, int) () at /lib/x86_64-linux-gnu/libQt5Quick.so.5
#4  0x00007fffed1795df in QQuickImageBase::load() () at /lib/x86_64-linux-gnu/libQt5Quick.so.5
#5  0x00007fffec5fe090 in QQmlPropertyPrivate::write(QObject*, QQmlPropertyData const&, QVariant const&, QQmlContextData*, QFlags<QQmlPropertyData::WriteFlag>) () at /lib/x86_64-linux-gnu/libQt5Qml.so.5
#6  0x00007fffec65bff8 in QQmlBinding::slowWrite(QQmlPropertyData const&, QQmlPropertyData const&, QV4::Value const&, bool, QFlags<QQmlPropertyData::WriteFlag>) () at /lib/x86_64-linux-gnu/libQt5Qml.so.5
#7  0x00007fffec65d738 in  () at /lib/x86_64-linux-gnu/libQt5Qml.so.5
#8  0x00007fffec65ee1e in  () at /lib/x86_64-linux-gnu/libQt5Qml.so.5
#9  0x00007fffec65ca14 in QQmlBinding::update(QFlags<QQmlPropertyData::WriteFlag>) () at /lib/x86_64-linux-gnu/libQt5Qml.so.5
#10 0x00007fffec63b67d in QQmlNotifier::emitNotify(QQmlNotifierEndpoint*, void**) () at /lib/x86_64-linux-gnu/libQt5Qml.so.5
#11 0x00007fffebfb7e75 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#12 0x00007fffebfb82e6 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#13 0x00005555556d25a7 in OCC::UserInfo::slotAvatarImage(QImage const&) ()
#14 0x00007fffebfb82e6 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#15 0x00007ffff7d8de36 in OCC::AvatarJob::avatarPixmap(QImage const&) () at /lib/x86_64-linux-gnu/libnextcloudsync.so.0
#16 0x00007ffff7dd7abe in OCC::AvatarJob::finished() () at /lib/x86_64-linux-gnu/libnextcloudsync.so.0
#17 0x00007ffff7dd2e1d in OCC::AbstractNetworkJob::slotFinished() () at /lib/x86_64-linux-gnu/libnextcloudsync.so.0
#18 0x00007fffebfb82e6 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#19 0x00007fffec2b9a07 in  () at /lib/x86_64-linux-gnu/libQt5Network.so.5
#20 0x00007fffebfae651 in QObject::event(QEvent*) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#21 0x00007ffff027d013 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#22 0x00007fffebf821ca in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#23 0x00007fffebf84bc1 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#24 0x00007fffebfda1c7 in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#25 0x00007ffff7c404db in g_main_context_dispatch () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#26 0x00007ffff7c40788 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#27 0x00007ffff7c40853 in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#28 0x00007fffebfd9843 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#29 0x00007fffebf80a4b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#30 0x00007fffebf88fc6 in QCoreApplication::exec() () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#31 0x00005555556149c9 in main ()
```

**Comment by er-vin** [comment]
Duplicate of #2710

**Comment by cybik** [comment]
To be fair, the fix in https://github.com/nextcloud/desktop/pull/2713 lets the app work, but the pop-up menu doesn't show the avatar image anymore.

Opening the settings shows it, however.

**Comment by er-vin** [comment]
> To be fair, the fix in #2713 lets the app work, but the pop-up menu doesn't show the avatar image anymore.

Sure I kind of expected something like that. I'm not sure why this started to happen though. Could you open a new issue for this please? It'd be especially interesting to get the debug logs.
